### PR TITLE
Add the kms-key-id parameter to oke config

### PR DESCRIPTION
**Pull request by jrosinsk**
This PR ddds support for creating an OKE cluster where the Kubernetes secrets are encrypted at rest in etcd using the privide kms vault master key.

```
resource "rancher2_cluster" "dev" {
     oke_config {
          ...
          kms_key_id = "ocid1.key.oc1.phx.<redacted>"
          ...
```

https://github.com/rancher-plugins/kontainer-engine-driver-oke/pull/41
https://github.com/rancher-plugins/kontainer-engine-driver-oke/issues/32

### CODE REVIEW

**Comment by jrosinsk** [comment]
This is merged and tagged in the driver.
https://github.com/rancher-plugins/kontainer-engine-driver-oke/pull/41

**Comment by rawmind0** [comment]
New OKE driver support not yet merged at Rancher v2.5, https://github.com/rancher/rancher/pull/33280 . This is required before merge this PR

**Comment by jrosinsk** [comment]
@rawmind0 , looks like the dependency was merged.

**Comment by rawmind0** [comment]
@jrosinsk , yes, PR was approved and merged 2 days ago, but it's not yet included at any Rancher release. It will be included at next Rancher v2.5.9.  Once Rancher v2.5.9 is released and PR https://github.com/rancher/terraform-provider-rancher2/pull/692 is updated and merged, we'll add this one.

**Comment by jrosinsk** [comment]
Got it, thanks for the explanation @rawmind0.

**Review by rawmind0 — APPROVED** [review]
LGTM

File: docs/resources/cluster.md [hunk 1]
```diff
@@ -1478,6 +1478,7 @@ The following arguments are supported:
 * `enable_private_nodes` - (Optional) Specifies whether worker nodes will be deployed into a new, private, subnet. Default `false` (bool)
 * `fingerprint` - (Required) The fingerprint corresponding to the specified user's private API Key (string)
 * `flex_ocpus` - (Optional) Specifies number of OCPUs for nodes (requires flexible shape specified with `node_shape`) (int)
+* `kms_key_id` - (Optional) The OCID of a KMS vault master key used to encrypt secrets at rest. See [here](https://docs.oracle.com/en-us/iaas/Content/ContEng/Tasks/contengencryptingdata.htm) for help creating a vault and master encryption key. Just for Rancher v2.5.9 or above (string)
```
Comment (rawmind0) [inline docs/resources/cluster.md:1481]: @jrosinsk with your permission, adding `Just for Rancher v2.5.9 or above` comment to clarify

**Review by rawmind0 — APPROVED** [review]
LGTM

File: docs/resources/cluster.md (hunk 1 above)
Comment (jrosinsk) [reply docs/resources/cluster.md:1481]: LGTM.

**Pull request merged**